### PR TITLE
fix: paginate charts in dashboard modals

### PR DESCRIPTION
**Pull request by ZeRego**
Relates to: [#12759](https://github.com/lightdash/lightdash/issues/12759) 

### Description:

Changes:
- update content endpoint to search by name
- replace useChartSummaries for `useChartSummariesV2` in dashboard modals

Screenshot 2024-12-06 at 14 29 25
Screenshot 2024-12-06 at 14 28 34
Screenshot 2024-12-06 at 14 28 26

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/hooks/useChartSummariesV2.ts [hunk 1]
```diff
@@ -3,28 +3,56 @@ import {
...
+        },
+        getNextPageParam: (lastPage) => {
+            if (lastPage.pagination) {
+                return lastPage.pagination.page <
+                    lastPage.pagination.totalPageCount
+                    ? lastPage.pagination.page + 1
+                    : undefined;
+            }
         },
+        ...infinityQueryOpts,
```
Comment (almeidabbm) [inline packages/frontend/src/hooks/useChartSummariesV2.ts:56]: 👨‍🍳 🤌

File: packages/frontend/src/components/DashboardTiles/TileForms/AddChartTilesModal.tsx [hunk 2]
```diff
@@ -75,8 +88,40 @@ const SelectItem = forwardRef<HTMLDivElement, ItemProps>(
...
+    // This ensures that previously fetched chart are preserved even when the search query changes.
+    // Uses 'uuid' to remove duplicates and maintain a consistent set of unique charts.
+    const [savedQueries, setSavedQueries] = useState<ChartContent[]>([]);
+    useEffect(() => {
+        const allPages = chartPages?.pages.map((p) => p.data).flat() ?? [];
+
+        setSavedQueries((previousState) =>
+            uniqBy([...previousState, ...allPages], 'uuid'),
+        );
+    }, [chartPages?.pages]);
```
Comment (almeidabbm) [inline packages/frontend/src/components/DashboardTiles/TileForms/AddChartTilesModal.tsx:124]: Couldn't this simply be a `useMemo`?

File: packages/frontend/src/components/DashboardTiles/TileForms/ChartUpdateModal.tsx [hunk 3]
```diff
@@ -50,8 +58,40 @@ const ChartUpdateModal = ({
...
+    // This ensures that previously fetched chart are preserved even when the search query changes.
+    // Uses 'uuid' to remove duplicates and maintain a consistent set of unique charts.
+    const [savedCharts, setSavedQueries] = useState<ChartContent[]>([]);
+    useEffect(() => {
+        const allPages = chartPages?.pages.map((p) => p.data).flat() ?? [];
+
+        setSavedQueries((previousState) =>
+            uniqBy([...previousState, ...allPages], 'uuid'),
+        );
+    }, [chartPages?.pages]);
```
Comment (almeidabbm) [inline packages/frontend/src/components/DashboardTiles/TileForms/ChartUpdateModal.tsx:94]: ditto

File: packages/frontend/src/components/DashboardTiles/TileForms/AddChartTilesModal.tsx (hunk 2 above)
Comment (ZeRego) [reply packages/frontend/src/components/DashboardTiles/TileForms/AddChartTilesModal.tsx:124]: We need the previous state to keep appending new results.

**Pull request merged**